bundles/org.openhab.ui/web/src/js/routes.js

```diff
@@ -77,6 +80,24 @@ const loadAsync = (page, props) => {
   }
 }
 
+const isLoggedIn = () => {
+  return store.getters.user !== null
+}
+
+const isAdmin = () => {
+  const user = store.getters.user
+  return user && user.roles && user.roles.indexOf('administrator') >= 0
+}
+
+const enforceAdmin = (to, from, resolve, reject) => {
+  if (!isAdmin()) {
+    reject()
+    authorize()
+  } else {
+    resolve()
+  }
+}
```

Comment:
Maybe those could be defined and exported & imported from elsewhere than routes.js (like auth.js) since they have less to deal with routes and more with authN?

Reply:
Agreed. I will move them around 👍